
message RequestIdentifier {
	required uint64 cookie = 1;
	required uint64 requestor_guid = 2;
	required string requestor_name = 3;	
}

message PlayerCommandMsg {

	message NewSongRequest {
		required string song_name = 1;
		optional uint32 position_in_ms = 2 [default=0];
	}

	required RequestIdentifier req_identifier = 1;
	optional bool stop_play = 2;
	optional NewSongRequest new_song_request = 3;
	
}

message PlayerCommandReplyMsg {
	required RequestIdentifier req_identifier = 1;

	// the status for the request for new song or new position
	required bool req_status = 2;
	required string req_status_desc = 3;

	// current status regardless of the request
	required bool is_song_playing = 4;
	optional string current_song_path = 5;
}